### PR TITLE
8306057: False arguments calling dispatch_base for aarch64

### PR DESCRIPTION
**Pull request by Kick-it11**
Hi,
  While I was reading the source codes, I found there might be a bug here.
   
  -- In interp_masm_aarch64.cpp, dispatch_base was define as
              void dispatch_base(TosState state,
                                              address* table,
                                              bool verifyoop = true,
                                              bool generate_poll = false);

  -- But in dispatch_next of interp_masm_aarch64.cpp,
     it is called:
           dispatch_base(state,
                                   Interpreter::dispatch_table(state),
                                   generate_poll);

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306057](https://bugs.openjdk.org/browse/JDK-8306057): False arguments calling dispatch_base for aarch64

### Reviewers
 * [Andrew Haley](https://openjdk.org/census#aph) (@theRealAph - **Reviewer**)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/12706/head:pull/12706` \
`$ git checkout pull/12706`

Update a local copy of the PR: \
`$ git checkout pull/12706` \
`$ git pull https://git.openjdk.org/jdk.git pull/12706/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12706`

View PR using the GUI difftool: \
`$ git pr show -t 12706`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12706.diff">https://git.openjdk.org/jdk/pull/12706.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/12706#issuecomment-1510773264)

### CODE REVIEW

File: src/hotspot/cpu/aarch64/interp_masm_aarch64.cpp [hunk 1]
```diff
@@ -515,7 +515,7 @@ void InterpreterMacroAssembler::dispatch_only_noverify(TosState state) {
 void InterpreterMacroAssembler::dispatch_next(TosState state, int step, bool generate_poll) {
   // load next bytecode
   ldrb(rscratch1, Address(pre(rbcp, step)));
-  dispatch_base(state, Interpreter::dispatch_table(state), generate_poll);
+  dispatch_base(state, Interpreter::dispatch_table(state), true, generate_poll);
```
Comment (theRealAph) [inline src/hotspot/cpu/aarch64/interp_masm_aarch64.cpp:518]:
```suggestion
  dispatch_base(state, Interpreter::dispatch_table(state), /*verifyoop*/true, generate_poll);
```

OK with this change. Thanks.

**Review by theRealAph — APPROVED** [review]

**Comment by Kick-it11** [comment]
@jaskarth That would be great! Thank you! In fact I do not know what kind of bug this is. I just happened to find this while I was reading the codes.

**Comment by jaskarth** [comment]
No problem! I've opened an issue for you: https://bugs.openjdk.org/browse/JDK-8306057
You can now update the title of the PR as `8306057: False arguments calling dispatch_base for aarch64` which will allow your PR to pass jcheck and complete tier 1 testing on Github Actions, as well as to integrate the changes so that they can be sponsored by a committer or reviewer.

**Review by RealFYang — APPROVED** [review]
I have also checked riscv port and looks like we have made it right :-)

**Pull request closed without merging**